### PR TITLE
zyre: migrate to Conan v2, drop older version

### PR DESCRIPTION
**Pull request by valgur**
TODO

- [ ] 
    > /usr/bin/ld: cannot find -lZeroMQ::libzmq
       /usr/bin/ld: cannot find -lczmq::czmq
- [ ] `ERROR: Missing prebuilt package for 'czmq/4.2.1'`

### CODE REVIEW

**Comment by valgur** [comment]
@uilianries `ERROR: Missing prebuilt package for 'czmq/4.2.1'`

**Comment by uilianries** [comment]
I just triggered a CI build to fix czmq/4.2.1

**Comment by uilianries** [comment]
Closing to re-create the CI build for this PR.

**Comment by valgur** [comment]
Closing temporarily to avoid unnecessary load on the CI. Will reopen when I'm actively working on the PR again.

**Review by uilianries — COMMENTED** [review]
Looking really good, just a comment that could improve a bit related to how zyre includes its dependencies. Maybe opening a PR to the upstream 🤔

File: recipes/zyre/all/patches/2.0.1-0001-cmake-use-conan-disable-executables.patch [hunk 1]
```diff
@@ -2,51 +2,7 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
...
--    include_directories(${LIBZMQ_INCLUDE_DIRS})
--    list(APPEND MORE_LIBRARIES ${LIBZMQ_LIBRARIES})
--  IF (PC_LIBZMQ_FOUND)
--      set(pkg_config_names_private "${pkg_config_names_private} libzmq")
--      list(APPEND OPTIONAL_LIBRARIES_STATIC ${PC_LIBZMQ_STATIC_LDFLAGS})
--  ELSE (PC_LIBZMQ_FOUND)
--      set(pkg_config_libs_private "${pkg_config_libs_private} -lzmq")
--  ENDIF (PC_LIBZMQ_FOUND)
--ELSE (LIBZMQ_FOUND)
-+find_package(ZeroMQ REQUIRED)
```
[the remark this answers is not in the document]
Comment (valgur) [reply recipes/zyre/all/patches/2.0.1-0001-cmake-use-conan-disable-executables.patch:20]: Uhm, the code you are commenting on is the old, removed one. I don't think it applies anymore?

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**